### DISCUSSION TITLE
How to create issues via an action

### DISCUSSION THREAD

**Discussion by iam-py-test** [community/community#25698] [Actions]
<p>I was trying to create an GitHub action (<a href="https://github.com/iam-py-test/my_filters_001/blob/e1b4ab04b56d51e09b25a19a648f7335b1ba9ecc/.github/workflows/main.yml" rel="noopener nofollow ugc">here</a>) that open an issue or at least adds a comment to an existing issue when an event occurs.<br>
However, whenever I try to do it using curl, it says <em>needs authentication</em>. I tried authenticating to my account via <code>${{ secrets.GITHUB_TOKEN }}</code> but I get an error.<br>
What am I doing wrong?<br>
I am not sure if this is the correct place to ask for help, but the docs did not provide any guidance.<br>
Is this require something special or can it just be done via normal actions or the REST api?</p>
<p>Thank you!</p>

**Comment by ghost** [comment, marked as answer]
<p>Hi, firstly we wanted to mention actions/github-script, as it might help you in the future to make API calls easier:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/actions/github-script">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/actions/github-script" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/10137/181092543-5172a6c0-4e44-4eb7-b348-3f9d0e180600.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/3/232e920e6d243c65138285738fea0e5a1a213ca9_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/3/232e920e6d243c65138285738fea0e5a1a213ca9_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/3/232e920e6d243c65138285738fea0e5a1a213ca9.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/3/232e920e6d243c65138285738fea0e5a1a213ca9_2_10x10.png"></div>

<h3><a href="https://github.com/actions/github-script" target="_blank" rel="noopener">actions/github-script</a></h3>

  <p>Write workflows scripting the GitHub API in JavaScript - actions/github-script</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>For the error itself, we weren’t able to find the specific workflow run demonstrating this error, but most likely you could resolve this issue by using a PAT instead of the GITHUB_TOKEN:</p>
<p><a href="https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token</a></p>
<p>You can then add it as a secret, and use the secret in the workflow to make the API call:</p>
<p><a href="https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow</a></p>

**Comment by iam-py-test** [comment]
<p><a class="mention" href="/github-support">@github-support</a> I am a bit busy, but I will look at that later.<br>
Thank you for your help!</p>

**Comment by iam-py-test** [comment]
<p>I have found a solution.<br>
Thanks!</p>

**Comment by dradetsky** [comment]
<p>another option: it appears that the runners have the <a href="https://cli.github.com/" rel="noopener nofollow ugc">github cli</a> installed on them <a href="https://github.com/dradetsky/misc-gha-tests/runs/3014549155?check_suite_focus=true" rel="noopener nofollow ugc">by default</a></p>

**Comment by iam-py-test** [comment]
<p><a class="mention" href="/dradetsky">@dradetsky</a> Thanks!<br>
I will look into GitHub cli!</p>
